error-issue/src/blame_finder.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {parsePrNumber, parseStacktrace} from './utils';
+import {RateLimitedGraphQL} from './rate_limited_graphql';
+import {BlameRange, GraphQLResponse, ILogger, StackFrame} from './types';
+
+/**
+ * Service for looking up Git blame info for lines in a stacktrace.
+ */
+export class BlameFinder {
+  // TODO(rcebulko): Use instance-level caching scheme.
+  private files: {[key: string]: Array<BlameRange>} = {};
+  private graphql: (query: string) => Promise<GraphQLResponse>;
+
+  constructor(
+    private repoOwner: string,
+    private repoName: string,
+    client: RateLimitedGraphQL,
+    private logger: ILogger = console
+  ) {
+    this.graphql = async query => client.runQuery(query);
+  }
+
+  /** Fetches the blame info for a file. */
+  async blameForFile(ref: string, path: string): Promise<Array<BlameRange>> {
+    const cacheKey = `${ref}:${path}`;
+    if (cacheKey in this.files) {
+      this.logger.info(`Returning blame for \`${cacheKey}\` from cache`);
+      return this.files[cacheKey];
+    }
+
+    const queryRef = async (ref: string): Promise<GraphQLResponse> => {
```

Comment:
Not sure what you mean by this

Reply:
Not sure what you mean by this.

Reply:
By moving data calls I mean moving `queryRef` into its own module.

Reply:
I don't feel that's necessary here. The query is a part of finding blame, that's what the class is meant to own. If there were a handful of different GraphQL operations taking place that would be one thing, but since it's just the one I'd prefer it live in this module.